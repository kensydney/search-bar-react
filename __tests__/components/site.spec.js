import React from 'react';
import {shallow} from 'enzyme';
import Site from '../../src/components/site';
import renderer from 'react-test-renderer';


const site = {
  "id": 2,
  "siteName": "Ebay",
  "siteUrl": "www.ebay.com.au",
  "description": "This is the description for ebay",
  "categoryIds": [
    1
  ]
};

describe('<Site />', () => {
  it('should show site url', () => {
    const wrapper = shallow(<Site site = {site} />);    
    expect(wrapper.find('a').text()).toBe(site.siteUrl);
  });
});

describe('Site Snapshot',()=>{
  it('capturing Snapshot of Site', () => {
    const renderedValue =  renderer.create(<Site site={site}/>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});