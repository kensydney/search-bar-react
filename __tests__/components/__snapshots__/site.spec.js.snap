// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site Snapshot capturing Snapshot of Site 1`] = `
<div
  className="inner-container site-content"
>
  <div>
    <a
      href="http://www.ebay.com.au"
    >
      www.ebay.com.au
    </a>
  </div>
  <div>
    This is the description for ebay
  </div>
</div>
`;
